pkg/endpoint/bpf.go

```diff
@@ -396,6 +396,16 @@ func (e *Endpoint) regenerateBPF(regenContext *regenerationContext) (revnum uint
 
 	// Skip BPF if the endpoint has no policy map
 	if e.isProperty(PropertySkipBPFPolicy) {
+		// Ingress endpoint needs entries in the endpoints map so that IPv6 ND works for it.
+		if e.isProperty(PropertyAtHostNS) {
+			stats.mapSync.Start()
+			err = lxcmap.WriteEndpoint(datapathRegenCtxt.epInfoCache)
+			stats.mapSync.End(err == nil)
+			if err != nil {
+				return 0, fmt.Errorf("Exposing endpoint in endpoints BPF map failed: %w", err)
+			}
+		}
+
```

Comment:
Maybe update the comment, since this isn't really specific to IPv6 ND anymore ?

This code path looks completely tuned to what we currently need for the Ingress endpoint :smile:. But abstracting it is a future problem ...

Reply:
Updated to:
```
		// Ingress endpoint needs entries in the endpoints map so that the return traffic,
		// ARP, and IPv6 ND are delivered to the host stack in all datapath configurations.
```